### PR TITLE
Calendar Validation

### PR DESCRIPTION
**Pull request by William-Unosquare**
- Validation before opening the booking modal.
  - Booking modal does not open if it's not part of the month being viewed on the calendar (to accompany disabled cursor and greyed out slots)
  - Booking modal does not open and a warning displays if a weekend is selected.
  - Booking modal does not open and a warning displays if attempting to book a new event in the past.

### CODE REVIEW

File: src/pages/Dashboard/container.js [hunk 1]
```diff
@@ -9,6 +9,10 @@ import eventsView from '../../utilities/eventsView';
 import moment from 'moment';
 import eventTypes from '../../utilities/eventTypes';
 import holidayStatus from '../../utilities/holidayStatus';
+import {checkIfPastDatesSelected,
+  checkIfDatesFallOnWeekend,
```
Comment (BrownMark) [inline src/pages/Dashboard/container.js:14]: It seems to be that Utilities is concerned with a number of different things. In this class alone I can see it supplying:
1. constants
2. extensions for toast
3. business logic

Not for this PR, but I think some of these things would be within services (business logic), some within constants (event types and holiday status) and the rest staying in utilities.

File: src/utilities/dashboardEvents.js [hunk 2]
```diff
@@ -159,6 +148,13 @@ export const checkIfDatesFallOnWeekend = (start, end) => {
   }
 };
 
+export const checkSameMonth = (dateOne, dateTwo) => {
```
Comment (BrownMark) [inline src/utilities/dashboardEvents.js]: This is more like a date extension than a dashboard event.

**Review by BrownMark — APPROVED** [review]
Approving with some tech debt pointed out.

File: src/pages/Dashboard/container.js (hunk 1 above)
Comment (William-Unosquare) [reply src/pages/Dashboard/container.js:14]: Yea I was discussing this with Kirsty last week. There's definitely some tech debt there to split things out.

File: src/utilities/dashboardEvents.js (hunk 2 above)
Comment (William-Unosquare) [reply src/utilities/dashboardEvents.js]: Moved to `dates.js`

**Pull request merged**